src/Uno.UI.RuntimeTests/Tests/Windows_Storage/Given_StorageFile_Native.base.cs

```diff
@@ -0,0 +1,261 @@
+﻿#nullable enable
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Windows.Storage;
+
+namespace Uno.UI.RuntimeTests.Tests.Windows_Storage
+{
+	public abstract class Given_StorageFile_Native_Base
+	{
+		protected abstract Task<StorageFolder> GetRootFolderAsync();
+
+		protected virtual Task CleanupRootFolderAsync() => Task.CompletedTask;
+
+		[TestMethod]
+		public async Task When_CreateFile_Name_Matches()
+		{
+			var rootFolder = await GetRootFolderAsync();
+			var fileName = GetRandomTextFileName();
+			StorageFile? createdFile = null;
+			try
+			{
+				createdFile = await rootFolder.CreateFileAsync(fileName);
+				Assert.AreEqual(fileName, createdFile.Name);
+			}
+			finally
+			{
+				if (createdFile != null)
+				{
+					await createdFile.DeleteAsync();
+				}
+				await CleanupRootFolderAsync();
+			}
+		}
+
+		[TestMethod]
+#if WINDOWS_UWP
+		[Ignore("On UWP, DisplayName sometimes returns the name with, and sometimes without the extension.")]
+#endif
+		public async Task When_CreateFile_DisplayName_Matches()
+		{
+			var rootFolder = await GetRootFolderAsync();
+			var fileName = GetRandomTextFileName();
+			var fileNameWithoutExtension = Path.GetFileNameWithoutExtension(fileName);
+			StorageFile? createdFile = null;
+			try
+			{
+				createdFile = await rootFolder.CreateFileAsync(fileName);
+				Assert.AreEqual(fileNameWithoutExtension, createdFile.DisplayName);
+			}
+			finally
+			{
+				if (createdFile != null)
+				{
+					await createdFile.DeleteAsync();
+				}
+				await CleanupRootFolderAsync();
+			}
+		}
+
+		[TestMethod]
+		public async Task When_CreateFile_Without_Extension()
+		{
+			var rootFolder = await GetRootFolderAsync();
+			var fileName = Path.GetFileNameWithoutExtension(GetRandomTextFileName());
+			StorageFile? createdFile = null;
+			try
+			{
+				createdFile = await rootFolder.CreateFileAsync(fileName);
+				Assert.AreEqual(fileName, createdFile.Name);
+				Assert.AreEqual(fileName, createdFile.DisplayName);
```

Comment:
May suffers the same problem as previous test.

Reply:
In this case it luckily does not have this issue, because the file name itself has no extension